utils/Azure/acrTools.ts

```diff
@@ -0,0 +1,277 @@
+ */
+export async function getRegistryTokens(registry: Registry): Promise<{ refreshToken: any, accessToken: any }> {
+    const subscription = getRegistrySubscription(registry);
+    const tenantId: string = subscription.tenantId;
+    let azureAccount: AzureAccount = AzureCredentialsManager.getInstance().getAccount();
+
+    const session: AzureSession = azureAccount.sessions.find((s, i, array) => s.tenantId.toLowerCase() === tenantId.toLowerCase());
+    const { accessToken } = await acquireARMToken(session);
+
+    //regenerates in case they have expired
+    if (accessToken) {
+        let refreshTokenACR;
+        let accessTokenACR;
+
+        await request.post('https://' + registry.loginServer + '/oauth2/exchange', {
+            form: {
+                grant_type: 'access_token',
+                service: registry.loginServer,
+                tenant: tenantId,
+                access_token: accessToken
+            }
+        }, (err, httpResponse, body) => {
+            if (body.length > 0) {
+                refreshTokenACR = JSON.parse(body).refresh_token;
+            } else {
+                return;
+            }
+        });
+
+        await request.post('https://' + registry.loginServer + '/oauth2/token', {
+            form: {
+                grant_type: 'refresh_token',
+                service: registry.loginServer,
+                scope: 'registry:catalog:*',
+                refresh_token: refreshTokenACR
+            }
+        }, (err, httpResponse, body) => {
+            if (body.length > 0) {
+                accessTokenACR = JSON.parse(body).access_token;
+            } else {
+                return;
+            }
+        });
+        if (refreshTokenACR && accessTokenACR) {
+            return { 'refreshToken': refreshTokenACR, 'accessToken': accessTokenACR };
+        }
+    }
+    vscode.window.showErrorMessage('Could not generate tokens');
+}
+
+export async function acquireARMToken(localSession: AzureSession): Promise<{ accessToken: string; }> {
+    return new Promise<{ accessToken: string; }>((resolve, reject) => {
+        const credentials: any = localSession.credentials;
+        const environment: any = localSession.environment;
+        // tslint:disable-next-line:no-function-expression // Grandfathered in
+        credentials.context.acquireToken(environment.activeDirectoryResourceId, credentials.username, credentials.clientId, function (err: any, result: { accessToken: string; }): void {
+            if (err) {
+                reject(err);
+            } else {
+                resolve({
+                    accessToken: result.accessToken
+                });
+            }
+        });
+    });
+}
+
+/**
+ *
+ * function used to create header for http request to acr
+ */
+export function getAuthorizationHeader(username: string, password: string): string {
+    let auth;
+    if (username === '00000000-0000-0000-0000-000000000000') {
+        auth = {
+            bearer: password
+        }
```

Comment:
the function return string. I think you need to return 

auth = `Bearer {password}`